w01/hhplus-tdd-jvm/src/main/kotlin/io/hhplus/tdd/point/PointService.kt

```diff
@@ -0,0 +1,31 @@
+package io.hhplus.tdd.point
+
+import io.hhplus.tdd.common.TimeUtil
+import io.hhplus.tdd.database.PointHistoryTable
+import io.hhplus.tdd.database.UserPointTable
+import org.springframework.stereotype.Service
+
+@Service
+class PointService(
+    private val userPointTable: UserPointTable,
+    private val pointHistoryTable: PointHistoryTable,
+    private val timeUtil: TimeUtil
+) {
+    fun charge(userId: Long, amount: Long): UserPoint {
+        val existingUserPoint = userPointTable.selectById(id = userId)
+        val totalAmount = existingUserPoint.point + amount
```

Comment:
```suggestion
        val chargedUserPoint = existingUserPoint.charge(amount)
        userPointTable.insertOrUpdate(userId, chargedUserPoint.point)
data class UserPoint(val id: Long, val point: Long, val updateMillis: Long) {
    fun charge(amount: Long): UserPoint {
        val total = this.point + amount
        require(total <= MAXIMUM_POINT) { "Total point must be less than $MAXIMUM_POINT" }
        return this.copy(point = total, updateMillis = System.currentTimeMillis())
    }
}
```

포인트 계산에 대한 로직은 `UserPoint`객체 내부에서 처리되면 어떨까요?